lib/FlagDependencyExportsPlugin.js

```diff
@@ -9,21 +9,24 @@ class FlagDependencyExportsPlugin {
 	apply(compiler) {
 		compiler.plugin("compilation", (compilation) => {
 			compilation.plugin("finish-modules", (modules) => {
-
 				const dependencies = Object.create(null);
 
 				let module;
 				let moduleWithExports;
+				let moduleProvidedExports;
 				const queue = modules.filter((m) => !m.providedExports);
 				for(let i = 0; i < queue.length; i++) {
 					module = queue[i];
 
 					if(module.providedExports !== true) {
 						moduleWithExports = false;
+						moduleProvidedExports = Array.isArray(module.providedExports) ? new Set(module.providedExports) : new Set();
 						processDependenciesBlock(module);
 						if(!moduleWithExports) {
 							module.providedExports = true;
 							notifyDependencies();
+						} else if(module.providedExports !== true) {
```

Comment:
Ok. I wonder if it would make sense to refactor that true state as an empty array. Possibly simpler code overall (one less type to worry about). That would be a separate PR, though.